modules/qr_code.py

```diff
@@ -0,0 +1,55 @@
+import qrcode
+import uuid
+import os
+import logging
+
+logger = logging.getLogger(__name__)
+
+class QRCode:
+    def __init__(self, base_url, qr_cache, max_size) -> None:
+        self.mapping = {}
+        self.base_url = base_url
+        self.qr_cache = qr_cache
+        self.max_size = max_size
+
+    def add(self, alias: str):
+        try:     
+            url = f"{self.base_url}/{alias}" 
+            filename = f"{uuid.uuid4()}.jpg"
+            path = os.path.join(self.qr_cache, filename)
+
+            img = qrcode.make(url)
+            img.save(path)
+            self.mapping[alias] = path
+
+            if len(self.mapping) > self.max_size: #removes files if exceeds size
+                remove_alias,remove_path = self.mapping.popitem()
+                os.remove(remove_path)
+                logger.debug(f"Removed qrcode with alias: {remove_alias} to free space.")
```

Comment:
we dont wanna exceed the size, so we can put this if statement before we create the image, and change the logic to 

```py
if len(self.mapping) >= self.max_size:
```

let me know what you think

Reply:
resolved by [565b354](https://github.com/SCE-Development/cleezy/pull/39/commits/565b354929418b5f1b12a691502e6bf365f3cee3)

Reply:
while the logic is correct, the if statement itself should go before we create the qr code. otherwise for a moment we are exceeding our cache size

say we have 100 files, we create 101 before we check if we should remove one from the cache